app/src/main/java/com/amaze/filemanager/adapters/AppsAdapter.java

```diff
@@ -249,9 +242,23 @@ public void onPositive(MaterialDialog materialDialog) {
             });
 
             popupMenu.inflate(R.menu.app_options);
+            if(!isGooglePlayStoreAvailable()) {
+                popupMenu.getMenu().findItem(R.id.play).setVisible(false);
+            }
             popupMenu.show();
         });
 
     }
 
+    private final boolean isGooglePlayStoreAvailable() {
+        PackageManager pm = context.getPackageManager();
+        try {
+            pm.getPackageInfo(COM_ANDROID_VENDING, PackageManager.GET_ACTIVITIES);
+            return true;
```

Comment:
When I test on emulator, the app crashes as before. I think the package is present but not the activity. Changing these lines to:
```java
    PackageInfo packageInfo = pm.getPackageInfo(COM_ANDROID_VENDING, PackageManager.GET_ACTIVITIES);
    return packageInfo.activities != null;
```
Seems to fix it.

Reply:
Being a cautious one I'd check both.